ffi/src/schema.rs

```diff
@@ -189,15 +261,14 @@ pub unsafe extern "C" fn visit_schema(
             DataType::Map(mt) => {
                 call!(
                     visit_map,
-                    // TODO: Should we no longer support value_contains_null argument in visitor_map
-                    // if nullability of the value will anyway be known through simple type visitor?
-                    // Same for contains_null in visitor_array.
-                    mt.value_contains_null,
                     visit_map_types(visitor, mt, mt.value_contains_null)
                 )
             }
             DataType::Array(at) => {
-                call!(visit_array, at.contains_null, visit_array_item(visitor, at, at.contains_null))
+                call!(
+                    visit_array,
+                    visit_array_item(visitor, at, at.contains_null)
+                )
```

Comment:
nit: more compact to factor out the nested function call:
```suggestion
                let child_list_id = visit_array_item(visitor, at, at.contains_null);
                call!(visit_array, child_list_id)
```